src/NATS.Client.Core/INatsConnection.cs

```diff
@@ -1,7 +1,24 @@
 namespace NATS.Client.Core;
 
-public interface INatsConnection
+public interface INatsConnection : IAsyncDisposable
```

Comment:
Why is it necessary to make the interface disposable? Is it because of a DI issue or to follow a general pattern of connection pooling as in database connections?

Reply:
I think that since only the creator of the concrete class should be disposing it, that the interface probably shouldn't be disposable?

Reply:
@mtmk @caleblloyd Example usages:

```async using INatsConnection con = new NatsConnection()```

```async using INatsConnection con = connectionPool.GetConnection()```

Reply:
@mtmk @caleblloyd our current v1 code does something like this:
```
public class NatsConnectionWrapper : IAsyncDisposable
{
    public INatsConnection Connection { get; }

    internal NatsConnectionWrapper(INatsConnection connection) //For Unit Testing Only
    {
        Connection = connection;
    }

    public NatsConnectionWrapper() : this(new NatsConnection()) { }

    public async ValueTask DisposeAsync() => await Connection.DisposeAsync().ConfigureAwait(false);
}
```

Reply:
consistency is good enough reason. I can't see a downside personally.